sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/client_base.py

```diff
@@ -5,58 +5,96 @@
 from typing import TYPE_CHECKING
 
 from azure.core.pipeline import Pipeline
-from azure.core.pipeline.policies import(
-    ContentDecodePolicy, UserAgentPolicy, DistributedTracingPolicy, HttpLoggingPolicy
+from azure.core.pipeline.policies import (
+    ContentDecodePolicy,
+    UserAgentPolicy,
+    DistributedTracingPolicy,
+    HttpLoggingPolicy,
 )
 from azure.core.pipeline.transport import RequestsTransport
-from ._generated import KeyVaultClient
+from ._generated.v7_0.version import VERSION as V7_0_VERSION
+from ._generated.v2016_10_01.version import VERSION as V2016_10_01_VERSION
 from .challenge_auth_policy import ChallengeAuthPolicy
 from .._user_agent import USER_AGENT
 
 if TYPE_CHECKING:
     # pylint:disable=unused-import
-    from typing import Any, Optional
+    from typing import Any, Optional, Union
     from azure.core.credentials import TokenCredential
     from azure.core.pipeline.transport import HttpTransport
     from azure.core.configuration import Configuration
-
-KEY_VAULT_SCOPE = "https://vault.azure.net/.default"
+    from ._generated.v7_0 import KeyVaultClient as KeyVaultClientV7
+    from ._generated.v2016_10_01 import KeyVaultClient as KeyVaultClientV2016_10_01
+
+    KeyVaultClientTypes = Union[KeyVaultClientV7, KeyVaultClientV2016_10_01]
+
+DEFAULT_API_VERSION = V7_0_VERSION
+
+def _get_configuration_class(api_version):
+    # type: (str) -> Configuration
+    """
+    Get the versioned configuration implementation corresponding to the current profile.
+    :return: The versioned configuration implementation.
+    """
+    if api_version == V7_0_VERSION:
+        from ._generated.v7_0._configuration import KeyVaultClientConfiguration as ImplConfig
+    elif api_version == V2016_10_01_VERSION:
+        from ._generated.v2016_10_01._configuration import KeyVaultClientConfiguration as ImplConfig
+    else:
+        raise NotImplementedError("API version {} is not supported by this package".format(api_version))
```

Comment:
This error message should match the async client's. Can't be an f-string here though 😞